tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -107,6 +107,13 @@ public void CloseSpan(Span span)
                 _spans.Add(span);
                 _openSpans--;
 
+                // If we close the local root span, make sure to unreference it in case we were to send another
+                // span in the same context later on (important in the case of AAS, to propagate resource.id)
+                if (span == RootSpan)
+                {
+                    RootSpan = default;
```

Comment:
Is it ok to clear `RootSpan`? Are we sure we don't use it anywhere else even after the root span is closed?

Reply:
Yeah I assume it's not (as I would have assumed this use case not to be possible either). So changing the implementation to follow zach's suggestion that is way better anyway